vermouth/dssp/dssp.py

```diff
@@ -194,10 +195,14 @@ def run_dssp(system, executable='dssp', savefile=None, defer_writing=True, versi
         Parse a DSSP output.
     """
     # check version
-    process = subprocess.run(["dssp", "--version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    version_found = process.stdout.decode('UTF8')
-    if version not in version_found:
-        raise DSSPError('Vermouth currently only supports DSSP version 3.0.0.')
+    supported_versions = [2, 3]
+    process = subprocess.run([executable, "--version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    match = re.search('\d+\.\d+\.\d+', process.stdout.decode('UTF8'))
+    version_found = match[0] if match else None
+    if not version_found:
+        raise DSSPError('Failed to get DSSP version information.')
+    if not int(version_found.split('.')[0]) in supported_versions and version != version_found:
+        raise DSSPError(f'DSSP {version_found} is not supported.')
```

Comment:
I have the feeling using something like https://packaging.pypa.io/en/latest/version.html#packaging.version.parse will result in simpler code. If you feel like over-engineering this, you can consider allowing more specific versions in SUPPORTED_DSSP_VERSIONS, but this is completely optional.

I'm also a bit confused on the meaning of `version` vs `supported_versions`, especially when I read the docstring. It seems they both cover the same use case. I'm not sure what the intended use was for the `version` argument, nor what it adds at the moment. It would be fine with me to remove the argument --- it's not used anywhere anyway. @fgrunewald You added this argument back in the day, is my interpretation correct?

Reply:
I kept `version` for the case that you want to support DSSP 3 only as a default.
In that case, `supported_versions` will contain only '3', and DSSP 2 can be supported by passing specific version with `version` argument.

Reply:
We implemented the version check as a quick fix, so I'm not surprised about the bug. In my opinion the executable thing is the most important to fix. 

We actually had a longer - though recorded - discussion about the version check. At the moment, we can only read DSSP 3 format correctly, I believe. Even worse, we only check and also in the CI test against DSSP 3.0.0 specifically. That's why we specifically search for this version string in the output instead of something fancier. Here we decided simpler would be better. We cannot know which versions are installed by users and a wrong dssp parsing is something, which is not easily caught. However, if it goes wrong it is very very bad for the protein model v3 and even more so v2. Also there are some issues with the version printing in some DSSP exe outputs, which is why the string search seemed safer.

As far as I see it, we have two options: 1) Simply fix the bug keep the version check. 2) Fix the version check but then also implement tests for all supported dssp versions incl. minor versions or specify minor versions as well. For sure we need to have tests then. 

The important question is: @juminlee what motivated you in the first place to change the version check? 

Also integration tests are failing, which indicates something goes wrong.

Reply:
Update: Locally the CI with this PR works for me. 🤔

Reply:
ok that's what I suspected. Can you run the integration tests using the following steps:

- if not installed, install pytest 
- in the GitHub repro go to vermouth-martinize/vermouth/tests/integration_tests
- run `pytest test_integration.py`

If it runs through, it probably means we are good parsing DSSP 2.2.1 files.

Reply:
DSSP 2.2.1 passed all the tests.
[test_vermouth.log](https://github.com/marrink-lab/vermouth-martinize/files/10875121/test_vermouth.log)

Reply:
So, what do we want to do?

Reply:
Unfortunately developing and maintaining martinize2 is no longer part of my day job :'( so I may not always respond as quickly as I would like.

I partly agree with @fgrunewald here, we /do/ need tests, but IMO it would suffice to test 1) the latest 2.x version; 2) the latest 3.x version; and 3) whatever version the CI uses (which is the version distributed with Ubuntu 20.04).
I don't think it's either feasible nor desirable to test much more. Would it be an option to generate the dssp output for those versions of dssp on a not too simple protein, and check whether the parsing works/produces the expected output?

The CI failure is *very* disturbing, it looks like there's missing backbone bonds? If it doesn't go away with the next commit one of us will have to dig into that, my guess would be a change in scipy's kdtree.

Reply:
Adding onto this discussion: I agree with @pckroon that we can have some simple tests, which only read the dssp output from two major versions. Instead of one big protein, however, I suggest we collect a bunch of fragments that cover the existing sec. structure assignments by dssp. 

@juminlee The fact that your CI tests worked, means that your executable is not linked against the modified vermouth-library as defined in this PR. I had the same problem initially. You need to link against the editable files using `pip install -e ./`. I'm not sure what the conda equivalent is. Can you try and see if you can reproduce, the failing CI with the current PR.

Reply:
Ok, executive decision:
1) We will extend dssp support to the latest bugfix releases of versions 2 and 3.
2) We will test the parsing of the output files of those versions
3) We will issue *warnings* for any other version of dssp people will try to use, rather than simply refusing to work altogether.
@juminlee Is this something you can work with?

Reply:
Are you suggesting to have a specific versions in `supported_versions`?
I can make it to contain ("2.2.1", "3.0.0"), but I do not have time to test other versions.

Reply:
For #3, I can change following line:
`raise DSSPError(f'DSSP {version_found} is not supported.')`

to:
`LOGGER.warning("Vermouth is tested only with DSSP versions {}. "`
`               "The specified DSSP version {} may cause unexpected results.",`
`               SUPPORTED_DSSP_VERSIONS, version_found,`
`               type='DSSP-version')`

However, does this mean that the `version` argument is no longer needed?

Reply:
I updated the code to issue a warning instead of raising an error 77c6a9933fd1d3b9e106ecae19bcd8d855ac9804.
In addition, I removed all version arguments.

> For point 2, it would entail running DSSP (verions 2.2.1 and 3.0.0) on a carefully selected PDB file, and recording the output. Then we can add a testcase that runs that output through read_dssp2, and making sure the secondary structure string is what we expect it to be.
The biggest issue would be getting an appropriate pdb file.

It would be great if someone could take over this.

Reply:
@fgrunewald It's been way too long since I touched a protein, can you (or someone in the lab) create an appropriate PDB file for me to play around with?
I think I prefer the idea of a crystal structure with a RCSB/PDB accession ID, that will make regenerating the whole thing easier later on, since that means we won't have to store the PDB file.

I can turn that into a test case or 2.